addons/binding/org.openhab.binding.irtrans/src/main/java/org/openhab/binding/irtrans/handler/EthernetBridgeHandler.java

```diff
@@ -0,0 +1,925 @@
+    public static final String REMOTE = "remote";
+
+    private static final Pattern RESPONSE_PATTERN = Pattern.compile("..(\\d{5}) (.*)", Pattern.DOTALL);
+    private static final Pattern HEX_PATTERN = Pattern.compile("RCV_HEX (.*)");
+    private static final Pattern IRDB_PATTERN = Pattern.compile("RCV_COM (.*),(.*),(.*),(.*)");
+
+    private Logger logger = LoggerFactory.getLogger(EthernetBridgeHandler.class);
+
+    private Selector selector;
+    private SocketChannel socketChannel;
+    protected SelectionKey socketChannelKey;
+    protected ServerSocketChannel listenerChannel;
+    protected SelectionKey listenerKey;
+    protected boolean previousConnectionState;
+    private final Lock lock = new ReentrantLock();
+
+    private List<TransceiverStatusListener> transceiverStatusListeners = new CopyOnWriteArrayList<>();
+
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Data structure to store the infrared commands that are 'loaded' from the
+     * configuration files. Command loading from pre-defined configuration files is not supported
+     * (anymore), but the code is maintained in case this functionality is re-added in the future
+     **/
+    protected final Collection<IrCommand> irCommands = new HashSet<IrCommand>();
+
+    public EthernetBridgeHandler(Bridge bridge) {
+        super(bridge);
+        // Nothing to do here
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing IRtrans Ethernet handler.");
+
+        // register ourselves as a Transceiver Status Listener
+        registerTransceiverStatusListener(this);
+
+        try {
+            selector = Selector.open();
+        } catch (IOException e) {
+            logger.error("An exception occurred while registering the selector: '{}'", e.getMessage());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR, e.getMessage());
+        }
+
+        if (selector != null) {
+            if (getConfig().get(IP_ADDRESS) != null && getConfig().get(PORT_NUMBER) != null) {
+                onUpdate();
+            } else {
+                logger.warn("Cannot connect to IRtrans Ethernet device. IP address or port number not set.");
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "IP address or port number not set.");
+            }
+
+            if (getConfig().get(IS_LISTENER) != null) {
+                configureListener((String) getConfig().get(LISTENER_PORT));
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Handler disposed.");
+
+        unregisterTransceiverStatusListener(this);
+
+        disconnect(socketChannel);
+
+        try {
+            if (selector != null) {
+                selector.close();
+            }
+        } catch (IOException e) {
+            logger.error("An exception occurred while closing the selector: '{}'", e.getMessage());
```

Comment:
Is this a bug situation if we get here, which the user should report?
If not, better reduce to debug